crates/wasi/src/preview2/error.rs

```diff
@@ -9,6 +9,20 @@ use std::marker;
 #[derive(Debug)]
 pub struct I32Exit(pub i32);
 
+impl I32Exit {
+    pub fn process_exit_code(e: &anyhow::Error) -> Option<i32> {
```

Comment:
Can you add docs here indicating what the return value means?

Reply:
Good catch. As I went to write docs I decided that downcasting and determining the exit code shouldnt be collapsed into a single associated function, and changed this to a `(&self) -> i32` that has a more straightforward doc, and inlined the downcasting at its 2 use sites.